main/integration-tests/__test__/api-tests/studies/study.test.js

```diff
@@ -0,0 +1,132 @@
+    it('should fail while trying to update Open Data studies', async () => {
+      // BUILD
+      // This is a known Open Data study
+      // but let the test admin confirm that anyway
+      const studyId = '1000-genomes';
+      const openDataStudies = await listStudies(testAdmin.axiosClient, 'Open Data');
+      const studyOfInterest = _.find(openDataStudies, study => study.id === studyId);
+      expect(studyOfInterest.category).toBe('Open Data');
+
+      const updateRequest = { id: studyId, description: 'Sample desc change' };
+
+      const nonAdminUser = await testFixture.createNonAdminUser(testAdmin.axiosClient, testProjectId);
+      const params = updateStudyParams(studyId, updateRequest);
+
+      // EXECUTE & CHECK
+      await expect(nonAdminUser.axiosClient.put(params.api, params.body)).rejects.toMatchObject({
+        response: { status: 404 },
+      });
+    });
+
+    it("should fail while trying to update another user's personal studies", async () => {
+      // BUILD
+
+      const nonAdminUser = await testFixture.createNonAdminUser(testAdmin.axiosClient, testProjectId);
+      const studyA = await testFixture.createMyStudy(nonAdminUser.axiosClient, testProjectId);
+
+      // UserB tries to access nonAdminUser's personal study
+      const userB = await testFixture.createNonAdminUser(testAdmin.axiosClient, testProjectId);
+      const updateRequest = { id: studyA.id, description: 'Sample desc change' };
+      const params = updateStudyParams(studyA.id, updateRequest);
+
+      // EXECUTE & CHECK
+      await expect(userB.axiosClient.get(params.api)).rejects.toMatchObject({
+        response: { status: 404 },
+      });
+    });
+  });
+
+  describe('Get Specific Study API: /api/studies/:id', () => {
+    it("should fail while trying to get another user's personal studies", async () => {
+      // BUILD
+      const nonAdminUser = await testFixture.createNonAdminUser(testAdmin.axiosClient, testProjectId);
+      const studyA = await testFixture.createMyStudy(nonAdminUser.axiosClient, testProjectId);
+
+      // UserB tries to access nonAdminUser's personal study
+      const userB = await testFixture.createNonAdminUser(testAdmin.axiosClient, testProjectId);
+      const params = getStudyParams(studyA.id);
+
+      // EXECUTE & CHECK
+      await expect(userB.axiosClient.get(params.api)).rejects.toMatchObject({
+        response: { status: 404 },
+      });
+    });
+
+    it('should pass while trying to get own studies', async () => {
+      // BUILD
+      const nonAdminUser = await testFixture.createNonAdminUser(testAdmin.axiosClient, testProjectId);
+      const studyA = await testFixture.createMyStudy(nonAdminUser.axiosClient, testProjectId);
+      const params = getStudyParams(studyA.id);
+
+      // EXECUTE & CHECK
+      const response = await nonAdminUser.axiosClient.get(params.api);
+      expect(response.data).toHaveProperty('projectId');
+      expect(response.data).toHaveProperty('rev');
```

Comment:
We should be checking that the fields in the response have the same values as `studyA`, not just checking their existence.

Also, it'd be better to use `.toMatchObject` instead of one assertion per field.

https://jestjs.io/docs/en/expect#tomatchobjectobject

Reply:
Yeah, I did consider using this. The response object has some random string attached to it, so it might need some changes. I'll work on using toMatchObject

Reply:
I've removed this test for now, still a work in progress.